schema/fields.go

```diff
@@ -337,12 +341,12 @@ type FieldBuilder struct {
 	Fields      []*Field
 }
 
-func (f *FieldBuilder) Build(forSearchIndex bool) (*Field, error) {
+func (f *FieldBuilder) Build(setSearchDefaults bool) (*Field, error) {
 	fieldType := ToFieldType(f.Type, f.Encoding, f.Format)
-	if forSearchIndex {
+	if setSearchDefaults {
 		// for search indexes, any field in schema is search indexable if it is not set explicitly.
 		// Similarly, we also tag it with sort if it is numeric.
-		if len(f.FieldName) > 0 && fieldType != ObjectType {
+		if len(f.FieldName) > 0 && (fieldType != ObjectType || (fieldType == ObjectType && len(f.Fields) == 0)) {
```

Comment:
Can you move `(fieldType != ObjectType || (fieldType == ObjectType && len(f.Fields) == 0))` into its own function. Its a bit unwieldy to read

Reply:
sure